src/components/Search/Filter.tsx

```diff
@@ -0,0 +1,298 @@
+      { label: 'compute', value: FilterByAccessOptions.Compute }
+    ]
+  },
+  {
+    id: 'filterSet',
+    label: 'Categories',
+    type: 'filterList',
+    options: [
+      {
+        label: 'automotive',
+        value: 'automotive'
+      },
+      {
+        label: 'manufacturing',
+        value: 'manufacturing'
+      },
+      {
+        label: 'text analysis',
+        value: 'textAnalysis'
+      },
+      {
+        label: 'finance',
+        value: 'finance'
+      }
+    ]
+  }
+]
+
+export const filterSets = {
+  automotive: [
+    'charging',
+    'ev',
+    'gx4m',
+    'mobility',
+    'moveid',
+    'parking',
+    'traffic'
+  ],
+  manufacturing: [
+    'euprogigant',
+    'industry40',
+    'manufacturing',
+    'predictive-maintenance'
+  ],
+  textAnalysis: ['library', 'ocr', 'text-analysis'],
+  finance: ['graphql']
+}
+
+const purgatoryFilterItem = { display: 'show purgatory ', value: 'purgatory' }
+
+export function getInitialFilters(
+  parsedUrlParams: queryString.ParsedQuery<string>,
+  filterIds: (keyof Filters)[]
+): Filters {
+  if (!parsedUrlParams || !filterIds) return
+
+  const initialFilters = {}
+  filterIds.forEach((id) =>
+    !parsedUrlParams?.[id]
+      ? (initialFilters[id] = [])
+      : Array.isArray(parsedUrlParams?.[id])
+      ? (initialFilters[id] = parsedUrlParams?.[id])
+      : (initialFilters[id] = [parsedUrlParams?.[id]])
+  )
+
+  return initialFilters as Filters
+}
+
+export default function Filter({
+  addFiltersToUrl,
+  showPurgatoryOption,
+  expanded,
+  className
+}: {
+  addFiltersToUrl?: boolean
+  showPurgatoryOption?: boolean
+  expanded?: boolean
+  className?: string
+}): ReactElement {
+  const { filters, setFilters, ignorePurgatory, setIgnorePurgatory } =
+    useFilter()
+
+  const router = useRouter()
+
+  const parsedUrl = queryString.parse(location.search, {
+    arrayFormat: 'separator'
+  })
+
+  useEffect(() => {
+    const initialFilters = getInitialFilters(
+      parsedUrl,
+      Object.keys(filters) as (keyof Filters)[]
+    )
+    setFilters(initialFilters)
+  }, [])
+
+  async function applyFilter(filter: string[], filterId: keyof Filters) {
+    if (addFiltersToUrl) {
+      let urlLocation = await addExistingParamsToUrl(location, [filterId])
+
+      if (filter.length > 0 && urlLocation.indexOf(filterId) === -1) {
+        const parsedFilter = filter.join(',')
+        urlLocation = `${urlLocation}&${filterId}=${parsedFilter}`
+      }
+
+      router.push(urlLocation)
+    }
+  }
+
+  async function handleSelectedFilter(value: string, filterId: keyof Filters) {
+    const updatedFilters = filters[filterId].includes(value)
+      ? { ...filters, [filterId]: filters[filterId].filter((e) => e !== value) }
+      : { ...filters, [filterId]: [...filters[filterId], value] }
+    setFilters(updatedFilters)
+
+    await applyFilter(updatedFilters[filterId], filterId)
+  }
+
+  async function applyClearFilter(addFiltersToUrl: boolean) {
```

Comment:
suggestion: just naming this `clearFilters` is maybe a bit easier for readability.